lib/git_ops/version.ex

```diff
@@ -139,6 +141,25 @@ defmodule GitOps.Version do
     versions_equal?(%{left | pre: []}, %{right | pre: []})
   end
 
-  defp parse(text), do: Version.parse(text)
-  defp parse!(text), do: Version.parse!(text)
+  defp parse("", text), do: Version.parse(text)
+
+  defp parse(prefix, text) do
+    if String.starts_with?(text, prefix) do
+      text
+      |> String.trim_leading(prefix)
+      |> Version.parse()
+    else
+      :error
+    end
+  end
+
+  defp parse!(prefix, text) do
+    case parse(prefix, text) do
+      {:ok, parsed} ->
+        parsed
+
+      :error ->
+        raise ArgumentError, "Expected: #{text} to be parseable as a version, but it was not."
```

Comment:
Will this end up `raise`ing if there's a non-version-related tag?

Reply:
Only if you use `parse!` but that is only used when we're sure its something we can parse. Mostly because we need the text version and the parsed version in different parts for different reasons, so sometimes we re-parse the version. Probably an inefficiency really, but it works well enough for now.